plugins/Marketplace/vue/src/InstallAllPaidPluginsButton/InstallAllPaidPluginsButton.vue

```diff
@@ -6,72 +6,100 @@
 -->
 
 <template>
-  <button
-    class="btn"
-    @click.prevent="onInstallAllPaidPlugins()"
-    :disabled="disabled"
-  >
-    <MatomoLoader v-if="loading" />
-    {{ translate('Marketplace_InstallPurchasedPlugins') }}
-  </button>
-  <div
-    class="ui-confirm"
-    id="installAllPaidPluginsAtOnce"
-    ref="installAllPaidPluginsAtOnce"
-  >
-    <h2>{{ translate('Marketplace_InstallAllPurchasedPlugins') }}</h2>
-    <p>
-      {{ translate('Marketplace_InstallThesePlugins') }}
-    </p>
-    <ul>
-      <li v-for="pluginDisplayName in paidPluginsToInstallAtOnce" :key="pluginDisplayName">
-        {{ pluginDisplayName }}
-      </li>
-    </ul>
-    <p>
-      <input
-        role="install"
-        type="button"
-        :data-href="installAllPaidPluginsLink"
-        :value="translate(
+  <div v-if="paidPluginsToInstallAtOnceArray.length">
+    <button
+      class="btn installAllPaidPluginsAtOnceButton"
+      @click.prevent="onInstallAllPaidPlugins()"
+      :disabled="disabled"
+    >
+      <MatomoLoader v-if="loading"/>
+      {{ translate('Marketplace_InstallPurchasedPlugins') }}
+    </button>
+    <div
+      class="ui-confirm"
+      id="installAllPaidPluginsAtOnce"
+      ref="installAllPaidPluginsAtOnce"
+    >
+      <h2>{{ translate('Marketplace_InstallAllPurchasedPlugins') }}</h2>
+      <p>
+        {{ translate('Marketplace_InstallThesePlugins') }}
+      </p>
+      <ul>
+        <li v-for="pluginDisplayName in paidPluginsToInstallAtOnceArray" :key="pluginDisplayName">
+          {{ pluginDisplayName }}
+        </li>
+      </ul>
+      <p>
+        <input
+          role="install"
+          type="button"
+          :data-href="installAllPaidPluginsLink"
+          :value="translate(
                   'Marketplace_InstallAllPurchasedPluginsAction',
-                  paidPluginsToInstallAtOnce.length,
+                  paidPluginsToInstallAtOnceArray.length,
                 )"
-      />
-      <input
-        role="cancel"
-        type="button"
-        :value="translate('General_Cancel')"
-      />
-    </p>
+        />
+        <input
+          role="cancel"
+          type="button"
+          :value="translate('General_Cancel')"
+        />
+      </p>
+    </div>
   </div>
 </template>
 
 <script lang="ts">
 import { defineComponent } from 'vue';
-import { Matomo, MatomoUrl, MatomoLoader } from 'CoreHome';
+import {
+  Matomo, MatomoUrl, MatomoLoader, AjaxHelper,
+} from 'CoreHome';
 
+interface installAllPaidPluginsButton {
+  paidPluginsToInstallAtOnceArray: Array<string>;
+  installNonceValue: string;
+}
 export default defineComponent({
   components: { MatomoLoader },
   props: {
     paidPluginsToInstallAtOnce: {
       type: Array,
-      required: true,
+      required: false,
+      default: () => [],
     },
     installNonce: {
       type: String,
-      required: true,
+      required: false,
     },
```

Comment:
I guess those props can now be removed, as they are unused? (And in the overview component, they now longer need to be passed in)

Reply:
Done :+1: